### ISSUE TITLE
`balena api-key generate` throws BalenaNotLoggedIn error, when logged in

### ISSUE THREAD

**Issue by srlowe** [balena-io/balena-cli#1748]
`balena api-key generate` throws BalenaNotLoggedIn error, even when logged in.  

It appears that the API route `https://api.balena-cloud.com/api-key/user/full` is returning a `401 Not Authorized` even though `Authorization` header is set correctly in request from SDK.

**Comment by srlowe** [comment]
Issue created on API repo: https://github.com/balena-io/balena-api/issues/2555

**Comment by pdcastro** [comment]
@srlowe, was this error also happening before the `balena-errors` / `balena-sdk` update? For example, with CLI v11.30.17.

**Comment by srlowe** [comment]
@pdcastro Yes. I've got a copy of v11.30.5 which I'm using to compare pre-conversion behaviour, and it already happens there.

**Comment by srlowe** [comment]
Looks like an API issue (I've opened a ticket there)

**Comment by srlowe** [comment]
Seems this is a permissions issue, depending on how authentication is done.  Looking at how we should handle this in the CLI.

See google chat:
> Pagan Gazzard: you need the auth.create_token permission for that, which only a user JWT has - api keys cannot create other full power api keys

**Comment by pdcastro** [comment]
@srlowe, now that you mentioned _"depending on how authentication is done,"_ it rang a bell. Logging in with a [session token](https://dashboard.balena-cloud.com/preferences/access-tokens) allows the command to work, but logging in with an API key does not:
```
$ balena login --token <very-long-session-token-string>
...
Successfully logged in as: gh_paulo_castro

$ balena api-key generate test-4921
Registered api key 'test-4921':
E7hGDZLFnzD5HOS7bX2t2ZGfLw5BE4y8

$ balena login --token E7hGDZLFnzD5HOS7bX2t2ZGfLw5BE4y8
...
Successfully logged in as: gh_paulo_castro


$ balena api-key generate test-7633
BalenaNotLoggedIn: You have to log in
```
Above:
* Logged in with session token
* Successfully generated an API key (`balena api-key generate test-4921`)
* Successfully used the generated API key to login again
* Failed to to generate another API key because I had logged in with an API key (instead of a session token)
* (I've already deleted the API key quoted in the test, no worries!)

The other day, someone else (can't remember who) had raised that some other command had failed and the reason was also that they had logged in with an API key, which didn't grant sufficient permissions for the required operation.

Perhaps for each command where we identify this situation (like `api-key generate`), we should check whether the user is logged in with an API key, and print an error that an API key is not good enough. This issue could be repurposed in that direction.
